### DISCUSSION TITLE
proposal: Make macOS check optional

### DISCUSSION THREAD

**Discussion by Xuanwo** [databendlabs/databend#5880] [Ideas]
Related issue: https://github.com/datafuselabs/databend/issues/5876

## Background

macOS build is the main bottleneck of our CI:

![image](https://user-images.githubusercontent.com/5351546/172974632-b8119186-a26f-41c6-ba48-2b7b4fb57196.png)

### macOS build is the slowest.

Ideally, our CI should finish within 30 minutes. But macOS may need up to 50 minutes to build. We need to find some way to address this issue.

![image](https://user-images.githubusercontent.com/5351546/172976181-8151d2c5-9b31-463a-ab9a-f6d1f44160a5.png)

### macOS build has the smallest quantity.

Compared to 30 Linux runners (20 hosted + 10 self-hosted), we only have 5 hosted mac runners. We often wait for macOS runners.

![image](https://user-images.githubusercontent.com/5351546/172976149-68f339a8-cc86-4c1a-94ee-f96588e45949.png)

## Proposal

So I propose to: make macOS check optional and allow PR get merged without waiting for macOS.

With this change, we can remove the `required` limit of `check`, so all our builds can happen concurrently.

For the bast case: We can make the developing CI finished inside **20 minutes**!

### Why not use self-hosted macOS runners?

It's expensive.

macOS runners are expensive (much more expensive than Linux instances, thanks to Apple)

It's hard to maintain.

macOS runners can't fit our existing runners' framework. We have to maintain macOS runners by ourselves. It's a huge burden for our developers.

Notice that we don't have a dedicated human to help maintain the runners. @everpcpc maintain them at his part-time job.

### Why not add an automated job for macOS build?

It's nice to have a `/run-macos-test` command or `macos-test` label to trigger macOS builds.

But we don't want to maintain too much automated infra, let's start with the simple way.

## Implementation

Implemented in https://github.com/datafuselabs/databend/pull/5884

**Comment by drmingdrmer** [comment]
Is the `Actions` tab the only place one could find the mac-os test result?

**Reply by Xuanwo** [reply]
I'm afraid so. Maybe we can place the test result badge in README.md? 

Like: ![image](https://user-images.githubusercontent.com/5351546/172978698-265558aa-3f62-419f-ab21-69891e340967.png)

**Comment by Xuanwo** [comment]
Advice from @zhang2014

We can allow PR to be merged without waiting macOS's CI results.
